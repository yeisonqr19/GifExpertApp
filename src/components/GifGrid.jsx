import React, { useState, useEffect } from "react";
import { getGifs } from "../helpers/GetGif";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then((gif) => setImages(gif));
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

export default GifGrid;
