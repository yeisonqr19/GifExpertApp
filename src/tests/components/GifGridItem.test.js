import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
describe("Test Para el componente GifGridItem", () => {
  const title = "Soy un nuevo titulo";
  const url = "https://localhost/imagen.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de Mostrar correctamente mi componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el titulo ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("la Imagen debe de tener el SRC y el ALT igual al de los props ", () => {
    const img = wrapper.find("img");
    // console.log(img.props().src);
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
});
