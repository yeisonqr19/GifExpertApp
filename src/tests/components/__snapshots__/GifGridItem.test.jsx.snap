// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Para el componente GifGridItem Debe de Mostrar correctamente mi componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Soy un nuevo titulo
  </p>
  <img
    alt="Soy un nuevo titulo"
    src="https://localhost/imagen.jpg"
  />
</div>
`;
